realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/SchemaTests.kt

```diff
@@ -0,0 +1,195 @@
+
+@RunWith(AndroidJUnit4::class)
+class SchemaTests {
+    @get:Rule
+    val configFactory = TestSyncConfigurationFactory()
+
+    @get:Rule
+    val errorCollector = ErrorCollector()
+
+    private lateinit var config: SyncConfiguration
+    private lateinit var app: TestRealmApp
+
+    @Before
+    fun setUp() {
+        app = TestRealmApp()
+        val user = createTestUser(app)
+        config = configFactory.createSyncConfigurationBuilder(user).build()
+    }
+
+    @After
+    fun tearDown() {
+        if (this::app.isInitialized) {
+            app.close()
+        }
+    }
+
+    @Test
+    fun instance() {
+        val realm = Realm.getInstance(config)
+        realm.use {
+            TestCase.assertFalse(realm.isClosed)
+        }
+        assertTrue(realm.isClosed)
+    }
+
+    @Test
+    fun createObject() {
+        Realm.getInstance(config).use { realm ->
+            realm.beginTransaction()
+            assertTrue(realm.schema.contains("StringOnly"))
+            val stringOnly = realm.createObject(StringOnly::class.java)
+            stringOnly.chars = "TEST"
+            realm.commitTransaction()
+            assertEquals(1, realm.where(StringOnly::class.java).count())
+        }
+    }
+
+    @Test
+    fun allow_createClass() {
+        DynamicRealm.getInstance(config).use { realm ->
+            val className = "Dogplace"
+            realm.beginTransaction()
+            realm.schema.create("Dogplace")
+            realm.commitTransaction()
+            assertTrue(realm.schema.contains(className))
+        }
+    }
+
+    @Test
+    fun allow_addField() {
+        // Init schema
+        Realm.getInstance(config).close()
+        val className = "StringOnly"
+        DynamicRealm.getInstance(config).use { realm ->
+            val objectSchema = realm.schema[className]
+            assertNotNull(objectSchema)
+            realm.beginTransaction()
+            objectSchema!!.addField("foo", String::class.java)
+            assertTrue(objectSchema.hasField("foo"))
+            realm.commitTransaction()
+            assertTrue(objectSchema.hasField("foo"))
+        }
+    }
+
+    // Special column "__OID" should be hidden from users.
+    @Test
+    fun fieldNames_stableIdColumnShouldBeHidden() {
+        val className = "StringOnly"
+        Realm.getInstance(config).use { realm ->
+            val objectSchema = realm.schema[className]
+            assertNotNull(objectSchema)
+            val names = objectSchema!!.fieldNames
+            assertEquals(1, names.size)
+            assertEquals(StringOnly.FIELD_CHARS, names.iterator().next())
+        }
+    }
+
+    enum class DestructiveSchemaOperation {
+        REMOVE_CLASS,
+        RENAME_CLASS,
+        SET_CLASS_NAME,
+        REMOVE_FIELD,
+        RENAME_FIELD,
+        REMOVE_INDEX,
+        REMOVE_PRIMARY_KEY,
+        ADD_PRIMARY_KEY,
+        ADD_FIELD_PRIMARY_KEY,
+    }
+
+    @Test
+    fun disallowDestructiveUpdateOfSyncedRealm() {
```

Comment:
Note, that the schema classes used by Realm/DynamicRealm are different, so these tests actually only test DynamicRealms. It is still valuable but would rename the test to `disallowDestructiveUpdateOfSyncedDynamicRealm`

Reply:
Ok